src/Silex/Provider/Form/SilexFormExtension.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/*
+ * This file is part of the Silex framework.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Silex\Provider\Form;
+
+use Silex\Application;
+use Symfony\Component\Form\Exception\InvalidArgumentException;
+use Symfony\Component\Form\FormExtensionInterface;
+use Symfony\Component\Form\FormTypeGuesserChain;
+
+class SilexFormExtension implements FormExtensionInterface
+{
+    private $silex;
+    private $types;
+    private $typeExtensions;
+    private $guessers;
+    private $guesserLoaded = false;
+    private $guesser;
+
+    public function __construct(Application $silex, array $types, array $typeExtensions, array $guessers)
+    {
+        $this->silex = $silex;
```

Comment:
Instances of `Silex\Application` are called `$app` elsewhere. 
The type hint could be done on `Pimple\Container` or even `ArrayAccess` for more reusability.